autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java

```diff
@@ -46,18 +47,31 @@ public MavenVersionRepository(String baseUrl) {
                     .collect(Collectors.toUnmodifiableSet());
         } catch (XmlPullParserException e) {
             throw new DependencyParserException("something went wrong while parseing the xml", e);
-        }catch (IOException e) {
+        } catch (IOException e) {
             return new HashSet<>();
         }
     }
 
-    private URL getRepoUri(Dependency dependency) {
+    private InputStream getInputStream(Dependency dependency) {
+        HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
HttpClient can be re-used by multiple threads. Might be best to define this as a field and re-use it for every call?